### PR TITLE
java 17 and compile issue with trs api

### PR DESCRIPTION
**Pull request by denis-yuen**
**Description**
Working on getting the nightly cron job to build properly post TRS update (but before avro revert)
Also updating to Java 17 with main debate in https://github.com/dockstore/dockstore/pull/5279
Coverage problem turned out to be a problem with jacoco's interaction with how surefire/failsafe fork VMs. Found a way to append arguments to the forked vm including jacoco without overwriting other custom arguments

Do not merge before https://github.com/dockstore/dockstore/pull/5279

**Review Instructions**
Builds including overnight build should work

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-4477 
Blocks https://ucsc-cgl.atlassian.net/browse/SEAB-5070
Created https://github.com/dockstore/dockstore/issues/5284 since system rules, junit4, seem on their last legs in Java 17
Created follow-up for quickstart and dockstore-documentation(?) https://ucsc-cgl.atlassian.net/browse/SEAB-5074

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `./mvnw clean install` 
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

**Comment by denis-yuen** [comment]
looking into coverage drop

**Comment by denis-yuen** [comment]
code scanning is a false positive created by pinning the java version

File: .github/workflows/mvnw.yml [hunk 1]
```diff
@@ -4,10 +4,10 @@ on: [push]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
     strategy:
       matrix:
-        java: [ '11', '17' ]
+        java: [ '17.0.4+8' ]
```
Comment (coverbeck) [inline .github/workflows/mvnw.yml:10]: Is it intentional this is different that the Circle CI version (17.0.3)?

**Review by coverbeck — APPROVED** [review]

File: .github/workflows/mvnw.yml (hunk 1 above)
Comment (denis-yuen) [reply .github/workflows/mvnw.yml:10]: No, I think this is just pinned to the same as in the webservice repo.

**Review by david4096 — APPROVED** [review]

**Pull request merged**